.github/workflows/publish.yml

```diff
@@ -28,25 +25,25 @@ jobs:
     - name: Build a binary wheel and a source tarball
       run: make build
 
-    # Publish to Test PyPI if this action was triggered by pushing a v* tag
-    - name: Publish 📦 to Test PyPI
-      if: github.event_name == 'push'  # check for `push` since action only triggered for tag pushes
-      run: |
-        python -m pip install .
-        echo "Dummy release of alibi-detect..."
-        python -c "from alibi_detect.version import __version__; print(__version__)" 
-      #     uses: pypa/gh-action-pypi-publish@release/v1
-      #     with:
-      #       password: ${{ secrets.TEST_PYPI_API_TOKEN }}
-      #       repository_url: https://test.pypi.org/legacy/
-
-    # If this action was triggered by a GitHub release, publish the associated tag to PyPI
+    # Get the PEP440 compliant version number (so we can check version is installable)
+    - name: Get 📦 version number
+      id: get_version
+      run: echo "PACKAGE_VERS=$(python setup.py --version)" >> $GITHUB_ENV
+
+    # Publish the associated tag to PyPI
     - name: Publish 📦 to PyPI
-      if: github.event_name == 'release'
+      uses: pypa/gh-action-pypi-publish@release/v1
+      with:
+        password: ${{ secrets.PYPI_API_TOKEN }}
+
+    # Wait for a while to give PyPI time to update (otherwise the pip install might fail)
+    - name: Sleep for 10 minutes
+      shell: bash
+      run: sleep 10m
```

Comment:
10 minutes sounds long, usually I can install almost straight away. Not sure though if worth trying programmatically installing (e.g. in a while loop) until it succeeds.

Reply:
Would a `while` loop not be essentially the same thing though? i.e. wouldn't we have to set some sort of break condition to prevent an infinite loop, so we'd still have to set a time or number of attempts limit? 

I've reduced it to 5 mins now. Perhaps we reduce it even more to say 1 or 2 mins, and then increase it if we regularly get failures? An occasional failure would not be the end of the world since the publish still would still have been done, and the user could check manually.

Reply:
Annoyingly 1 minute wasn't enough!
https://github.com/ascillitoe/alibi-detect/actions/runs/3496674516/jobs/5854858932

I've settled on 3 for now... we can add a fetch a cup of tea 🍵 stage to the instructions?!